File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
   public:
    string encode(vector<string>& strs) {
        string result;

        for (int i = 0; i < static_cast<int>(strs.size()); ++i) {
            result += to_string(static_cast<int>(strs[i].size()));
            result += '#';
            result += strs[i];
        }

        return result;
    }

    vector<string> decode(string s) {
        vector<string> result;
        string temp;

        for (int i = 0; i < static_cast<int>(s.size()); ++i) {
            if (s[i] == '#') {
                int const length = stoi(temp);
                temp.clear();  // clean up the number string

                for (int j = i + 1; j <= i + length; ++j) {
                    temp.push_back(s[j]);
                }

                result.push_back(temp);  // stores temp(the string we just read)
                temp.clear();            // clean up temp because we need to read the next number
                i += length;             // advance the iterator

                continue;
            }

            temp.push_back(s[i]);
        }

        return result;
    }
};

File: Data Structures & Algorithms/valid-sudoku/submission-1.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        s=set()

        for i in range(9):
            for j in range(9):
                if board[i][j]=='.':
                    continue
                if board[i][j] not in s:
                    s.add(board[i][j])
                    continue
                return False
            s.clear()

        for i in range(9):
            for j in range(9):
                if board[j][i]=='.':
                    continue
                if board[j][i] not in s:
                    s.add(board[j][i])
                    continue
                return False
            s.clear()

        for i in range(0, 9, 3):
            for j in range(0, 9, 3):
                for x in range(i, i+3):
                    for y in range(j, j+3):
                        if board[x][y] == '.':
                            continue
                        if board[x][y] not in s:
                            s.add(board[x][y])
                            continue
                        return False
                s.clear()
                    
        return True